generators/app/files/CONTRIBUTING.md

```diff
@@ -2,9 +2,123 @@
+and so that you stay in closer sync with the trunk.
+
+Remember to *frequently* [rebase](http://git-scm.com/book/ch3-6.html) your branch,
+to get the latest changes from `master`. This will give you access to any new
+features and utilities, plus make merging upstream (much) easier.
+
+### Naming
+
+Branch names should be:
+
+- Descriptive and concise (pithy, even)
+- All lowercase
+- Separated by *underscores*
+
+| Bad       | Good                |
+|-----------|---------------------|
+| `pnx`     | `init_phoenix`      |
+| `logo`    | `logo_v2`           |
+| `linter`  | `ci_linter_task`    |
+| `sockets` | `setup_web_sockets` |
+| `#42`     | `switch_to_adga`    |
+
+### Long-Running & Integration Branches
+
+In the edge case that a series of features are tightly coupled, and it is impractical to decouple them:
+
+1. Create a long-running "integration branch" from `master`
+2. Branch sub-features from the integration branch
+  - *Do not commit directly into the long-running branch directly*
+3. Frequently rebase sub-branches from the integration branch, and the feature branch from `master`
+4. When the sub-feature is complete, pull request against integration
+5. When the all coupled features are complete and merged into the integration branch,
+   pull request the integration branch against `master`
+
+It may be helpful to create a checklist in the integration branch's description
+of the subtasks that belong to it.
+
+### Commit Messages
+
+Provide a summary message of fewer than 50 characters. If further detail is required,
+provide a bullet-point list (or paragraph) on the line(s) below.
+
 ## Pull Requests
 
-TODO: Explain pull request process.
+Before a branch gets merged into `master`, it *must* be reviewed by another team member.
+This has a number of benefits:
+
+- Exposes the team to new code
+  - Keeps everyone in the loop
+  - Improves the project's [bus factor](https://en.wikipedia.org/wiki/Bus_factor)
+- Catch bugs early on
+- Lowers the risk of feature or utility duplication
+- Style consistency
+
+### Titles
+
+When creating your Pull Request, provide a short and descriptive title (fewer than 50 characters).
+These will likely be close to the branch name. Format like a normal sentence.
+
+| Bad         | Good                           |
+|-------------|--------------------------------|
+| `Quick fix` | `Add User model typespecs`     |
+| `#42`       | `Reduce Sass deep nesting`     |
+| `BUTTONS`   | `Update contact buttons to v3` |
+
+#### Descriptions
+
+To help the reviewer understand what they're reviewing, provide a description of what
+the code is expected to do. Link to any material, documentation, or resources that
+may be helpful for context.
+
+As a preemptive measure, be sure to mention any gotchas, or areas that may need explanation.
+
+#### Link Issues
+
+In GitHub, when you mention an Issue in a comment or description, it creates a link to and from the PR.
```

Comment:
GitHub Issues are capitalized, as it's a product
> GitHub’s tracker is called Issues, and has its own section in every repository.

So, it depends if we want to talk about "Issues" (GitHub), or "issues" (general)